nettoyage.py

```diff
@@ -408,5 +387,153 @@ def replace_char(df):
+                        col += "Modification"
+                if col not in col_add: # Cas ou on tombe sur le premier marche qui modifie un champ
+                    df[col] = "" # Initialisation dans le df initial
+                    col_add += [col]
+                df[col][i] = json_modification[0][col_init]
+                df["booleanModification"][i] = 1 #Création d'une booléenne pour simplifier le subset pour la suite
+    return df
+
+## Objectif numéro 3: Définition des fonctions nécéssaire pour l'identification des marchés/mis à jour de marchés
+
+def split_dataframe(df, sub_data, modalite):
+    """Définition de deux dataFrame.
+        - Le premier qui contiendra uniquement les lignes avec modification, pour le marché ayant pour objet modalite
+        - Le second contiendra l'ensemble des lignes correspondant au marché isolé dans le df1 qui ont pour objet modalite"""
+    #Premier df: Contenant les lignes d'un marche aec des colonnes modifications non vide
+    marche = sub_data[sub_data.objet == modalite]
+    marche = marche.sort_values(by = 'id')
+    #Second dataframe: Dans le df complet, récupération des lignes correspondant au marché récupéré
+    date = marche.datePublicationDonnees.iloc[0]
+    # A concaténer ? 
+    marche_init = df[df.objet == modalite]
+    marche_init = marche_init[marche_init.datePublicationDonnees == date]
+    return (marche, marche_init)
+    
+
+# Partie fusion des datas modifiées
+def fusion_source_modification(raw, df_source, col_modification, dict_modification):
+    """ Permet de fusionner les colonnes xxxModification et sa colonne """
+    for col in col_modification:
+        col_init = dict_modification[col]
+        df_source[col_init].loc[raw.name] = raw[col]
+        return df_source
+
+#Fonction Finale
+
+def regroupement_marche(df, dict_modification):
+    """Permet de recoder la variable identifiant. 
+    Actuellement: 1 identifiant par déclaration (marché initial / modification sur marché/ un marché peut être déclaré plusieurs fois en fonction du nombre d'entreprise)
+    En sortie: idMarche correspondra à un identifiant unique pour toutes les lignes composants un marché SI il a eu une modification"""
+    df["idtech"] = ""
+    col_modification = list(dict_modification.keys())
+    subdata_modif = df[df.booleanModification == 1] # Tout les marchés avec les modifications
+    liste_objet = list(subdata_modif.objet.unique())
+    df_to_concatene = pd.DataFrame() #df vide pour la concaténation
+    logger.info("Au total, {} marchés sont concernés par au moins une modification".format(len(liste_objet)))
+    for objet_marche in liste_objet:
+        #Récupération du dataframe modification et du dataframe source
+        marche, marche_init = split_dataframe(df, subdata_modif, objet_marche)
+        for j in range(len(marche)):
+            raw = marche.iloc[j]
+            marche_init = fusion_source_modification(raw, marche_init, col_modification, dict_modification)
```

Comment:
```suggestion
            marche_init = fusion_source_modification(raw, marche_init, dict_modification.keys(), dict_modification)
```
Ne pas passer par une variable intermédiaire est ici plus explicite.